src/utils/emitter.ts

```diff
@@ -0,0 +1,14 @@
+/**
```

Comment:
Should we maybe rename this file to `OpenCloseComponent`?

If Franco approves of this class, lets also add some doc for the methods `onOpen/onBeforeOpen...`